### DISCUSSION TITLE
'<' and '>' in server password produce invalid settings.xml

### DISCUSSION THREAD

**Discussion by alsha** [whelk-io/maven-settings-xml-action#155] [Q&A]
Hi,
i use following config:

```
      - name: Configure Maven settings.xml
        uses: whelk-io/maven-settings-xml-action@v21
        with:
          servers: >
            [{
               "id": "snapshot",
               "username": "${{ secrets.ARTIFACTORY_USER }}",
               "password": "${{ secrets.ARTIFACTORY_PASSWORD }}"
             }]
```
My secrets.ARTIFACTORY_PASSWORD looks something like this: "se&lt;cr&gt;et".  In this case invalid settings.xml is produced. 
'<' and '>'  must be escaped by the action implementation or CDATA must be used.

How can i workaround this issue?

Many thanks in advance!

**Comment by zteater** [comment]
@alsha you should remove special characters from your password. What you are facing is a [limitation with XML](https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references#List_of_predefined_entities_in_XML). 

You could also try the encoded values, like `&gt;` and `&lt;`, but really you should change your password to not include the XML special characters `<`, `>`, `&`, `"`, `'`.

**Comment by alsha** [comment]
@zteater sometimes the passwords are given you from outside, you can not change them...
By the way, here is the right solution of a problem: https://stackoverflow.com/questions/1091945/what-characters-do-i-need-to-escape-in-xml-documents
